src/H5Lmodule.h

```diff
@@ -30,7 +30,69 @@
  * Navigate back: \ref index "Main" / \ref UG
  * <hr>
  *
- * @todo Under Construction
+ * \section sec_link The HDF5 Link Interface
```

Comment:
Consistency in Terminology:
    - Some modules use "Interface" (e.g., "HDF5 Link Interface") while others use "APIs" (e.g., "HDF5 Table APIs")
    - Suggestion: Standardize terminology across all modules

Reply:
This would go beyond the scope of the PR, file a separate issue to standardize terminology across all the docs.